resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -696,18 +721,34 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
         }
         executorTerminated(d, slaveId, taskId, s"Executor finished with state $state")
         // In case we'd rejected everything before but have now lost a node
-        metricsSource.recordRevive
-        d.reviveOffers
+        reviveMesosOffers(Option(d))
+        logInfo("Reviving offers due to a finished executor task.")
```

Comment:
Put the log line before the `reviveMesosOffers` call. Also, as mentioned above, let's just revive if the executor failed and not if it's completed successfully.

Reply:
done